controllers/podcasts.go

```diff
@@ -0,0 +1,121 @@
+package controllers
+
+import (
+	"log"
+	"net/http"
+	"strconv"
+
+	"github.com/gorilla/mux"
+
+	"github.com/UniversityRadioYork/2016-site/models"
+	"github.com/UniversityRadioYork/2016-site/structs"
+	"github.com/UniversityRadioYork/2016-site/utils"
+	"github.com/UniversityRadioYork/myradio-go"
+)
+
+// PodcastController is the controller for the URYPlayer Podcast pages.
+type PodcastController struct {
+	Controller
+}
+
+// NewPodcastController returns a new PodcastController with the MyRadio session s
+// and configuration context c.
+func NewPodcastController(s *myradio.Session, c *structs.Config) *PodcastController {
+	return &PodcastController{Controller{session: s, config: c}}
+}
+
+// GetAllPodcasts handles the HTTP GET request r for the all postcasts page, writing to w.
+func (podcastsC *PodcastController) GetAllPodcasts(w http.ResponseWriter, r *http.Request) {
+
+	podcastm := models.NewPodcastModel(podcastsC.session)
+
+	podcasts, err := podcastm.GetAllPodcasts()
+
+	if err != nil {
+		//@TODO: Do something proper here, render 404 or something
+		log.Println(err)
+		return
+	}
+
+	data := struct {
+		Podcasts []myradio.Podcast
+	}{
+		Podcasts: podcasts,
+	}
+
+	err = utils.RenderTemplate(w, podcastsC.config.PageContext, data, "podcasts.tmpl")
+	if err != nil {
+		log.Println(err)
+		return
+	}
+
+}
+
+// Get handles the HTTP GET request r for a singular podcast page, writing to w.
+func (podcastsC *PodcastController) Get(w http.ResponseWriter, r *http.Request) {
+
+	podcastm := models.NewPodcastModel(podcastsC.session)
+
+	vars := mux.Vars(r)
+
+	id, _ := strconv.Atoi(vars["id"])
```

Comment:
what if id is not provided, or non-numeric?

Reply:
It may be my misunderstanding, but the server.go doesn't match to this function if it's not both of those, it's kinda regex.